qt/python/mantidqt/project/projectparser_mantidplot.py

```diff
@@ -0,0 +1,246 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+import re
+import matplotlib as mpl
+
+SCALE_OPTIONS = {0: "linear", 1: "log"}
+
+
+class MantidPlotProjectParser(object):
+    plot_layer_pattern = r"<multiLayer>\s*(.*?)\s*<\/multiLayer>"
+    graph_pattern = r"<graph>\s*(.*?)\s*<\/graph>"
+    workspaces_pattern = r"<mantidworkspaces>\s*(.*?)\s*<\/mantidworkspaces>"
+
+    def __init__(self, filename):
+        self.filename = filename
+        self.graph_text = []
+        self.layer_text = []
+        self.number_of_figures = 0
+        self.number_of_graphs = []
+        self._load_file_text()
+
+    def _load_file_text(self):
+        with open(self.filename, encoding="utf8") as project_file:
+            self.text = project_file.read()
+        self.layer_text = re.findall(self.plot_layer_pattern, self.text, re.DOTALL)
+        self.number_of_figures = len(self.layer_text)
+        self.number_of_graphs = [0] * self.number_of_figures
+
+    def get_workspaces(self):
+        """
+        Get workspaces to load from project file
+        """
+        ws_match = re.search(self.workspaces_pattern, self.text, re.DOTALL)
+        if ws_match:
+            # split by tab
+            mantid_workspace_entries = ws_match.group(1).split('\t')
+            if len(mantid_workspace_entries) > 1 and mantid_workspace_entries[0] == "WorkspaceNames":
+                # Load workspace groups
+                return self._load_workspaces(mantid_workspace_entries[1:])
+            else:
+                return []
+
+    def _load_workspaces(self, workspace_entries):
+        """
+        Identify workspaces within a <mantidworkspaces> tag
+        :param workspace_entries: The entries within the mantidworkspaces tag
+        :return list of workspaces found within the <mantidworkspaces> tag
+        """
+        ws_list = []
+        ws_group_members = []
+        for ws_entry in workspace_entries:
+            # if the entry within the <mantidworkspaces> contains a "," it defines a workspace group
+            if ',' in ws_entry:
+                ws_group = ws_entry.split(',')
+                # First entry is group name, remainder of the line contains members of the group
+                ws_group_members.extend(ws_group[1:])
+            else:
+                if ws_entry not in ws_group_members:
+                    ws_list.append(ws_entry)
+        return ws_group_members + ws_list
+
+    def get_plots(self):
+        """
+        Get plots to load from project file
+        """
+        plot_list = []
+        for i in range(self.number_of_figures):
```

Comment:
Just an FYI (but not worth holding up for) this can be rewritten as list comprehension 
`return [self._get_plot_creation_arguments(i) for i in range(self.number_of_figures)]`